### DISCUSSION TITLE
UDF not working in Duck db

### DISCUSSION THREAD

**Discussion by akash-goel** [duckdb/duckdb#6566] [General]
Hi Team ,

I am trying to create the UDF and run on Duck db using sample code . But i am getting error that **"UDF column type mismatch, expected [NULL, NULL], got [VARCHAR, VARCHAR] "** Can you please guide how can we write a sample udf function and run on the duckdb database.

`import duckdb
import pandas

df = pandas.DataFrame(['x@example.com', 'y@example.org'], columns=['email'])

def get_domain(df):
	df['domain'] = df['email'].apply(lambda x : x.split('@')[1]).to_frame()
	return df

rel = duckdb.from_df(df).map(get_domain)
print(rel.to_df())
`

**Comment by Tishj** [comment]
A recent issue also encountered this problem, I have pushed a fix in <https://github.com/duckdb/duckdb/pull/6571>
Let me know if that fixes your issue as well :)

**Reply by Tishj** [reply]
Actually, it doesn't because the amount of columns and the original columns differ.
As a work around, try returning some bogus data if:
```py
if len(df) == 0:
    # return bogus data that helps to infer the returned type
```
